packages/latex/src/main.rs

```diff
@@ -0,0 +1,222 @@
+use std::{
+    env,
+    fmt::Write,
+    io::{self, Read},
+};
+
+fn main() {
+    let args: Vec<String> = env::args().skip(1).collect();
+
+    let Some(action) = args.get(0) else {
+        eprintln!("No action was provided.");
+        return;
+    };
+
+    match action.as_str() {
+        "manifest" => print!("{}", &manifest()),
+        "transform" => {
+            let from = args.get(1).unwrap();
+            let format = args.get(2).unwrap();
+
+            if "latex" != format {
+                eprintln!("Output format not supported");
+                return;
+            }
+
+            print!("{}", transform(from));
+        }
+        other => eprintln!("Invalid action '{other}'"),
+    }
+}
+
+fn transform(from: &str) -> String {
+    let input: Value = {
+        let mut buffer = String::new();
+        io::stdin().read_to_string(&mut buffer).unwrap();
+        from_str(&buffer).unwrap()
+    };
+
+    match from {
+        "__bold" => transform_tag(input, "textbf"),
+        "__italic" => transform_tag(input, "emph"),
+        "__superscript" => transform_tag(input, "textsuperscript"),
+        "__subscript" => transform_tag(input, "textsubscript"),
+        "__underlined" => transform_tag(input, "underline"),
+        "__strikethrough" => transform_tag(input, "sout"), //fixme: needs a package to use
+        "__verbatim" => transform_block(input, "verbatim"),
+        "__paragraph" => transform_paragraph(input),
+        "__document" => transform_block(input, "document"),
+        "__text" => escape_text(input),
+        "__heading" => transform_heading(input),
+        _ => panic!("element not supported"),
+    }
+}
+
+fn transform_paragraph(paragraph: Value) -> String {
+    let mut result = String::new();
+    result.push('[');
+    write!(result, r#"{{"name": "raw", "data": "\n"}},"#,).unwrap();
+    if let Value::Array(children) = &paragraph["children"] {
+        for child in children {
+            result.push_str(&serde_json::to_string(child).unwrap());
+            result.push(',');
+        }
+    }
+    write!(result, r#"{{"name": "raw", "data": "\n"}}"#,).unwrap();
+    result.push(']');
+
+    result
+}
+
+fn transform_tag(node: Value, latex_function: &str) -> String {
+    let mut result = String::new();
+    result.push('[');
+    write!(result, r#"{{"name": "raw", "data": "\\{latex_function}{{"}},"#,).unwrap();
+    if let Value::Array(children) = &node["children"] {
+        for child in children {
+            result.push_str(&serde_json::to_string(child).unwrap());
+            result.push(',');
+        }
+    }
+    write!(result, r#"{{"name": "raw", "data": "}}"}}"#,).unwrap();
+    result.push(']');
+
+    result
+}
+
+fn transform_heading(heading: Value) -> String {
+    let mut result = String::new();
+    result.push('[');
+
+    let Value::String(s) = &heading["arguments"]["level"] else {
+        panic!();
+    };
+    let level = s.parse::<u8>().unwrap().clamp(1, 6);
```

Comment:
I think LaTeX (by default) only has `section`, `subsection` and `subsubsection`. Clamping your level to `6` will be able to generate `subsubsubsubsubsection`, which doesn't exist. It is possible to create more of such commands, so you will either need to include such commands (complicated, maybe bloated and maybe not so useful, especially without env variables) or clamp to 3.